modules/internal/CPtr.chpl

```diff
@@ -353,7 +372,7 @@ module CPtr {
 
     :arg data: the c_ptr to memory that was allocated
     */
-  inline proc c_free(data: c_ptr) {
+  inline proc c_free(data: c_void_ptr) {
```

Comment:
I think we still need the old c_free and that you should copy the function to make a version that applies to c_void_ptr. (or make it generic with `where isAnyCPtr(t)`). As far as I know we don't have any implicit conversions from c_ptr to c_void_ptr.

Reply:
Sure. Still, [this line of my test](https://github.com/chapel-lang/chapel/pull/7254/files/6dbcf375317047aa5165187ee1b81a3824add12a#diff-50dd7781e51e1b51a1c34b0fef8ddb8fR21) works and made it seem like `c_ptr` gets implicitly converted to `c_void_ptr`. Plus, [`c_free` is a pass-through call to `chpl_here_free`](https://github.com/chapel-lang/chapel/blob/master/modules/internal/CPtr.chpl#L356) (which accepts `c_void_ptr`), so the conversion seems to happen implicitly.

I like the `isAnyCPtr(t)` approach.

Reply:
I stand corrected - Elliot added implicit conversions from c_ptr to c_void_ptr in PR #3371. So, I don't think you need to add the 2nd method. Probably best to leave it alone (unless you really want the isAnyCPtr approach).

Reply:
I suppose that from a documentation point of view, the where clause is clearer than just accepting a c_void_ptr. If we leave it as-is, we probably should add a doc-comment that indicates the method will accept c_ptr in addition to c_void_ptr.